### PR TITLE
Add quickstart for function_calling_config.

### PR DESCRIPTION
**Pull request by markmcd**
This is a standalone guide for using the constrained decoding feature of function calling.

It's meant to be as simple as possible and only focuses on the `function_calling_config` feature, and is not meant to be a comprehensive guide to function calling (we have that already).

### CODE REVIEW

**Comment by markmcd** [comment]
NBReview here: https://app.reviewnb.com/google-gemini/cookbook/pull/31/

File: quickstarts/Function_calling_config.ipynb [hunk 1]
```diff
@@ -0,0 +1,269 @@
+{
```
Comment (pcoet) [inline quickstarts/Function_calling_config.ipynb:1]: <p>"none have been": I think there's a word missing here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

**Review by pcoet — APPROVED** [review]

File: quickstarts/Function_calling_config.ipynb (hunk 1 above)
Comment (random-forests) [inline quickstarts/Function_calling_config.ipynb:1]: <p>Nit: In the set_light_color function, maybe change the output to:</p><p> </p><p>print("Light-color set to: ", rgb_hex)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (random-forests) [inline quickstarts/Function_calling_config.ipynb:1]: <p>Idea, want to include an automatic function calling example after this block? Maybe something like:</p><p> </p><p>Text cell:</p><p> </p><p>## Automatic function calling</p><p>You can also enable automatic function calling in the Python SDK.</p><p> </p><p>Code cell:</p><p> </p><p> </p><p><code>chat = model.start_chat(enable_automatic_function_calling=True)</code></p><p><code>response = chat.send_message("Make this place PURPLE!", tool_config=tool_config)</code></p><p> </p><p>And should we include a "Learning more" section at the end? Maybe...</p><p> </p><p>## Learning more</p><p> </p><p>Check out the function calling [quickstart](<a href="https://github.com/google-gemini/cookbook/blob/main/quickstarts/Function_calling.ipynb)" target="_blank">https://github.com/google-gemini/cookbook/blob/main/quickstarts/Function_calling.ipynb)</a> for an introduction to function calling. You can find another fun function calling example [here](<a href="https://github.com/google-gemini/cookbook/blob/main/quickstarts/rest/Function_calling_REST.ipynb)" target="_blank">https://github.com/google-gemini/cookbook/blob/main/quickstarts/rest/Function_calling_REST.ipynb)</a> using <code>curl</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Review by MarkDaoust — COMMENTED** [review]
LGTM.

**Review by random-forests — APPROVED** [review]
Approving to unblock

File: quickstarts/Function_calling_config.ipynb [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        "def stop_lights():\n",
+        "    \"\"\"Stop flashing lights.\"\"\"\n",
+        "    print(\"Lights turned off.\")\n",
+        "\n",
+        "\n",
+        "light_controls = [enable_lights, set_light_color, stop_lights]\n",
+        "\n",
+        "model = genai.GenerativeModel(\"models/gemini-1.5-pro-latest\", tools=light_controls)\n",
+        "\n",
+        "history = [\n",
```
[the remark this answers is not in the document]
Comment (markmcd) [reply quickstarts/Function_calling_config.ipynb:90]: Yeah it is - the system instruction wasn't working yesterday, but that looks resolved now, I'll try again.

File: quickstarts/Function_calling_config.ipynb (hunk 1 above)
Comment (markmcd) [reply quickstarts/Function_calling_config.ipynb:1]: Done - also added `LIGHTBOT: ` prefixes to make it clear the output is from the functions, vs model response.

Comment (markmcd) [reply quickstarts/Function_calling_config.ipynb:1]: Done - it's very simple as we can't edit `tool_config` between automatic function calls yet but it illustrates the point..

File: quickstarts/Function_calling_config.ipynb [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+      "outputs": [],
+      "source": [
+        "def enable_lights():\n",
+        "    \"\"\"Turn on the lighting system.\"\"\"\n",
+        "    print(\"Lights enabled.\")\n",
+        "\n",
+        "\n",
+        "def set_light_color(rgb_hex: str):\n",
+        "    \"\"\"Set the light color. Lights must be enabled for this to work.\"\"\"\n",
+        "    print(\"Light-blinking enabled.\")\n",
```
[the remark this answers is not in the document]
Comment (markmcd) [reply quickstarts/Function_calling_config.ipynb]: ty!

**Pull request merged**